src/ui/auth/rackspace/rackspace.go

```diff
@@ -0,0 +1,174 @@
+type Auth struct{}
+
+// AuthResponse is the response from Rackspace Managed Kubernetes Auth (kubernetes-auth)
+// If the user is authenticated, the User struct is filled. Otherwise, the Error string is filled
+type AuthResponse struct {
+	APIVersion string `json:"apiVersion"`
+	Kind       string `json:"kind"`
+	Status     struct {
+		Authenticated bool   `json:"authenticated"`
+		Error         string `json:"error,omitempty"`
+		User          struct {
+			Username string              `json:"username"`
+			UID      string              `json:"uid,omitempty"`
+			Groups   []string            `json:"groups,omitempty"`
+			Extra    map[string][]string `json:"extra,omitempty"`
+		} `json:"user,omitempty"`
+	} `json:"status"`
+}
+
+var (
+	rackspaceMK8SAuthURLTokenEndpoint string
+)
+
+// Authenticate checks user's credential against the Rackspace Managed Kubernetes Auth (kubernetes-auth)
+// if the check is successful a dummy record will be inserted into DB, such that this user can
+// be associated to other entities in the system.
+func (l *Auth) Authenticate(m models.AuthModel) (*models.User, error) {
+
+	// kubernetes-auth only uses the token (m.Password) for auth. The username (m.Principal) isn't used at all.
+	// In fact, a user could put anything at all into the username field. It must be ignored.
+	// However, we log the username to help track the request because we can't put the token in the logs.
+	log.Debugf("ProvidedUsername=%s Authentication attempt", m.Principal)
+
+	// build auth request
+	authRequestBody := fmt.Sprintf("{\"spec\": {\"token\": \"%s\"}}", m.Password)
+	req, err := http.NewRequest(http.MethodPost, rackspaceMK8SAuthURLTokenEndpoint, strings.NewReader(authRequestBody))
+	if err != nil {
+		log.Errorf("ProvidedUsername=%s Error=%v", m.Principal, err)
+		return nil, err
+	}
+
+	log.Debugf("ProvidedUsername=%s Sending auth request to %s", m.Principal, rackspaceMK8SAuthURLTokenEndpoint)
+
+	// send auth request
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		log.Errorf("ProvidedUsername=%s Error=%v", m.Principal, err)
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	// read auth response body
+	authRespBody, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		log.Errorf("ProvidedUsername=%s Error=%v", m.Principal, err)
+		return nil, err
+	}
+
+	// check for auth server error
+	if resp.StatusCode >= http.StatusInternalServerError {
+		log.Errorf("ProvidedUsername=%s Error=%s", m.Principal, authRespBody)
+		return nil, fmt.Errorf("%d %s", resp.StatusCode, authRespBody)
+	}
+
+	// read auth response body as json
+	authResp := AuthResponse{}
+	err = json.Unmarshal([]byte(authRespBody), &authResp)
+	if err != nil {
+		log.Errorf("ProvidedUsername=%s Error=%v", m.Principal, err)
+		return nil, err
+	}
+
+	log.Debugf("ProvidedUsername=%s Authenticated=%t", m.Principal, authResp.Status.Authenticated)
+
+	// check for any status other than OK
+	if resp.StatusCode != http.StatusOK {
```

Comment:
why check for non-200 errors here? can we merge this check with lines 92+? I think any non-200 should return before we begin serializing JSON